src/ngx_http_lua_semaphore.c

```diff
@@ -0,0 +1,519 @@
+
+/*
+ * Copyright (C) cuiweixie
```

Comment:
@cuiweixie Will you please state that "I hereby assign copyright in this code to the lua-nginx-module project, to be licensed under the same terms as the rest of the code"? This is just for legal resolution. Thanks for your understanding.